ompi/mca/coll/han/coll_han_alltoallv.c

```diff
@@ -281,13 +374,15 @@ static int alltoallv_sendrecv_w(
            there is no more data coming, we could cancel the recvs we posted
         2) We could make a first-pass and count total bytes to recv, and be
            careful not to post more than we know is coming.
-        We take path 2 here, to avoid possible race conditions between the
-        cancel and the possibility of a matching recv.
+
+        After realizing that the convertor may decide to not fully pack the
```

Comment:
Slightly confused about the meaning of this ? The convertor might decide to avoid stopping in the middle of a predefined type. Is this what this comment is about?

Reply:
yes, that's what I intended to say.  I observed a strange type consisting of 1 char and 1 int, and the convertor refused to use the last 3 bytes of the buffer, which threw off my original counting scheme.